src/lib/components/EventSchedule/index.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../EventScheduleModal';
+
+/**
+ * @fileoverview A schedule manager which opens and closes modals based on
+ * the current href. Looks in children for content.
+ *
+ * This does not inherit from BaseStateElement as it is not a LitElement.
+ */
+
+class EventSchedule extends HTMLElement {
+  constructor() {
+    super();
+
+    this._modalElement = document.createElement('web-event-schedule-modal');
+    this._modalElement.className = 'web-modal';
+    this._modalElement.open = false;
+    this._modalElement.addEventListener('close-modal', () => {
+      if (!window.location.hash.substr(1)) {
+        return; // we closed ourselves, so got an event that can be ignored
+      }
+      const url = window.location.pathname + window.location.search;
+      window.history.replaceState(null, null, url);
+      this.onHashChange();
+    });
+
+    this._currentSession = null;
+    this.onHashChange = this.onHashChange.bind(this);
+    this.onClick = this.onClick.bind(this);
+  }
+
+  _elementForHash(hash) {
+    const id = hash.substr(1);
+    return (id && this.querySelector(`[data-session-id="${id}"]`)) || null;
+  }
+
+  /**
+   * Controls the open/close of session modals, by matching the page's hash to any contained
+   * elements with a matching `data-session-id` attribute.
+   *
+   * If no hash is set, or no element is found, hides any modal.
+   */
+  onHashChange() {
+    const session = this.isConnected
+      ? this._elementForHash(window.location.hash)
+      : null;
+    if (session === this._currentSession) {
+      return;
+    }
+
+    this._currentSession = session;
+    if (!session) {
+      this._modalElement.open = false;
+      this._modalElement.remove();
```

Comment:
Pressing the close button on the modal causes us to go down this code path. Normally the modal [waits for animationEnd to restore focus to the triggering element](https://github.com/GoogleChrome/web.dev/blob/master/src/lib/components/BaseModalElement/index.js#L134). Since we remove it right away, the focus blurs and goes back to `body`. It's not a great experience for a screen reader because they'll have to navigate the page all over again.

Do we need to remove the modal element or could we just let it handle its own close behavior?

Reply:
I fixed this. Basically we need to listen to `animationend` as you mentioned. I'm worried it's a bit flaky but eh.